### DISCUSSION TITLE
How to Sanitize mols when SanitizeMol return“ AtomKekulizeException: non-ring atom 6 marked aromatic”

### DISCUSSION THREAD

**Discussion by jingxingzhi** [rdkit/rdkit#4202] [Q&A]
I was trying to do an list of reactions for a mol,but the product is incorrectly labeled with aromaticity which cannot be dealed by Chem.SanitizeMol and Chem.Kekulize
```
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem import AllChem
rxn = AllChem.ReactionFromSmarts('[#6:1]1:[#6:2]:[#6:3]:[#6:4]:[#6:5]:[#6:6]:1>>[#6:1]1=[#6:2]-[#6:3]=[#6:4]-1.[#6:6]#[#6:5]')
mol=Chem.MolFromSmiles('COc1ccc2cc3[n+](cc2c1OC)CCc1cc(O)c(O)cc1-3')
reactants = (abc,)
res=rxn.RunReactants(reactants)
Chem.SanitizeMol(res[0][0])
#Chem.Kekulize(a,clearAromaticFlags=True)#will get the same result
```
which return 
> AtomKekulizeException                     Traceback (most recent call last)
> <ipython-input-48-d9caf75279bb> in <module>
>       3 a=res[0][0]
>       4 print(Chem.MolToSmiles(a))
> ----> 5 Chem.Kekulize(a,clearAromaticFlags=True)
>       6 print(Chem.MolToSmiles(a))
>       7 Chem.SanitizeMol(res[0][0])
> 
> AtomKekulizeException: non-ring atom 6 marked aromatic

the smile of res[0][0] was 
`'COc1ccc2cc3[n+](cc2c1OC)CCc1cc(O)c(O)cc1-3' `
wich should be 
`'C[N+]1C(CC2=CC=C2OC)-c2cc(O)c(O)cc2CC1'`
How can I solve this problem

**Comment by KKLagoda** [comment]
Having the same problem, where upon a break of a bond in a polycyclic aromatic system, some, but not all atoms become aliphatic. I tried specifying the product side in all possible ways to indicate some atoms turning aliphatic, but the properties do not seem to be taken into account.

Things I have tried:
- Specifying atoms becoming aliphatic with capital letters on the product side
- Indicating their hydrogen count on the product side
- adding queries like A, !a and !R on the product side

**Comment by ptosco** [comment]
Regarding the issue described in the original post, I believe the reaction needs to specify that the bond that becomes a triple bond should not originally be part of multiple rings, as a triple bond could not form within another 6-membered ring.
If you change the reaction to:
```
rxn = AllChem.ReactionFromSmarts('[#6:1]1:[#6:2]:[#6:3]:[#6:4]:[#6R1:5]:[#6R1:6]:1>>[C:1]1=[C:2]-[C:3]=[C:4]-1.[C:6]#[C:5]')
```
then you will get sensible products.
On top of that, you may re-set aromaticity flags based on the sizes of the rings in the products, to avoid that antiaromatic rings are still labelled as aromatic.
```
from rdkit import Chem
from rdkit.Chem import Draw
from rdkit.Chem import AllChem
import itertools

rxn = AllChem.ReactionFromSmarts('[#6:1]1:[#6:2]:[#6:3]:[#6:4]:[#6R1:5]:[#6R1:6]:1>>[C:1]1=[C:2]-[C:3]=[C:4]-1.[C:6]#[C:5]')
rxn
```
![image](https://github.com/rdkit/rdkit/assets/5244385/0768d41e-9ccb-42a7-9f5c-1a8f1f646f62)

```
mol = Chem.MolFromSmiles('COc1ccc2cc3[n+](cc2c1OC)CCc1cc(O)c(O)cc1-3')
mol
```
![image](https://github.com/rdkit/rdkit/assets/5244385/a96de9e5-607e-4113-afcf-457f8ca832f9)

```
res = rxn.RunReactants((mol,))
for prod in res:
    for p in prod:
        ri = p.GetRingInfo()
        atom_rings = ri.AtomRings()
        bond_rings = ri.BondRings()
        for ring_idx, atom_ring in enumerate(atom_rings):
            bond_ring = bond_rings[ring_idx]
            is_all_sp2_ring = all(p.GetAtomWithIdx(ai).GetHybridization() == Chem.HybridizationType.SP2 for ai in atom_ring)
            n_pi_e = len(atom_ring)
            is_aromatic = (n_pi_e - 2 % 4 == 0)
            for ai in atom_ring:
                p.GetAtomWithIdx(ai).SetIsAromatic(is_aromatic)
            for bi in bond_ring:
                p.GetBondWithIdx(ai).SetIsAromatic(is_aromatic)
        for a in p.GetAtoms():
            if not a.IsInRing():
                a.SetIsAromatic(False)
        for b in p.GetBonds():
            if not b.IsInRing():
                b.SetIsAromatic(False)
        Chem.SanitizeMol(p)
Draw.MolsToGridImage(tuple(itertools.chain.from_iterable(res)), molsPerRow=6)```
```
![image](https://github.com/rdkit/rdkit/assets/5244385/441373a7-e8e3-49e2-a315-1133ca9814e4)
